ark/nodes.py

```diff
@@ -252,6 +255,11 @@ def get_class_list(self) -> list[str]:
         return filters.apply('class_list', class_list, self)
 
 
+# Should we generate HTML for this node?
+def _generate_html(node):
+    return os.path.isfile(node.filepath) or site.config.get("dir_nodes", True)
```

Comment:
I'd like to make this configuration option as explicit and self-documenting as possible. What do you think of `"emit_no_file_directory_nodes"`, with a default of `True`?